### PR TITLE
 initial distribution of agents depends on the size of agents

### PR DESCRIPTION
**Pull request by xuqiancheng**
#420
Now initial distribution of agents depends on the size of agents.
But when the size of agents is normal distribution instead of a constant value, overlapping in initial distribution can still be observed.
May be we can introduce something like buffer distance between agents when doing initial distributing to avoid overlapping.  

Also make some changes to compile jpscore on windows with VS2017:
- VS2017 use OpenMP 2.0, so index variable in OpenMP 'for' statement must have signed integral type
- Add some code in CmakeLists.txt files for using MSVC
- The input argument of function 'void Configuration::SetTrajectoriesFile(std::string)' should be a std::sting instead of std::filesystem::path

### CODE REVIEW

File: pedestrian/AgentsParameters.cpp [hunk 1]
```diff
@@ -432,5 +432,13 @@ std::string AgentsParameters::writeParameter()
     return s;
 }
 
+double AgentsParameters::GetAminMean() {
+	return _Amin.mean();
```
Comment (planospeedo) [inline pedestrian/AgentsParameters.cpp]: spaces not tabs

File: pedestrian/AgentsParameters.cpp [hunk 2]
```diff
@@ -432,5 +432,13 @@ std::string AgentsParameters::writeParameter()
...
+}
+
+double AgentsParameters::GetBmaxMean() {
+	return _Bmax.mean();
```
Comment (planospeedo) [inline pedestrian/AgentsParameters.cpp]: spaces not tabs

File: pedestrian/AgentsSourcesManager.cpp [hunk 3]
```diff
@@ -214,7 +214,9 @@ void AgentsSourcesManager::ComputeBestPositionCompleteRandom(AgentsSource* src,
 {
      auto dist = src->GetStartDistribution();
      auto subroom = _building->GetRoom(dist->GetRoomId())->GetSubRoom(dist->GetSubroomID());
-     vector<Point> positions = PedDistributor::PossiblePositions(*subroom);
+     auto config=GetBuilding()->GetConfig();
+     auto distributor = std::unique_ptr<PedDistributor>(new PedDistributor(config));
+     vector<Point> positions = distributor->PossiblePositions(*subroom);
```
Comment (planospeedo) [inline pedestrian/AgentsSourcesManager.cpp]: Why was this changed?

Comment (xuqiancheng) [reply pedestrian/AgentsSourcesManager.cpp]: > Why was this changed?

Because I have changed the function PossiblePosition from static to non-static.
May be there are better way to doi it.
